frontend/src/app/components/ExperimentPage.js

```diff
@@ -382,11 +382,56 @@ export class ExperimentDetail extends React.Component {
   }
 
   onScroll() {
-    const sy = this.props.getScrollY();
-    this.setState({ useStickyHeader: sy > CHANGE_HEADER_ON });
+    const { getElementOffsetHeight, getScrollY } = this.props;
+    const sy = getScrollY();
+    const detailsHeaderWrapperHeight = getElementOffsetHeight('.details-header-wrapper');
+    const changeHeaderOn = this.setChangeHeaderOn(detailsHeaderWrapperHeight);
+    const useStickyHeader = sy > changeHeaderOn;
+    const stickyHeaderSiblingHeight = this.setStickyHeaderSiblingHeight(detailsHeaderWrapperHeight, useStickyHeader);
+
+    this.setState({
+      changeHeaderOn,
+      useStickyHeader,
+      stickyHeaderSiblingHeight });
     this.didScroll = false;
   }
 
+  // if there is no content in the .status-bar
+  // adjust the initial snap position and the header isn't yet sticky, adjust
+  // changeHeaderOn to the top of the .details-header
+  // instead of the bottom of the #main-header
+  setChangeHeaderOn(detailsHeaderWrapperHeight) {
+    const { getElementOffsetHeight } = this.props;
+    const statusBar = document.querySelector('.status-bar');
```

Comment:
You shouldn't need to query the DOM for `.status-bar` here. The logic for displaying the status bar [is here with the `statusType` variable](https://github.com/johngruen/testpilot/blob/3cacd7f58e27eba0a255951dd40042847bd2d857/frontend/src/app/components/ExperimentPage.js#L173) - we might want to split that out into a function to reuse it. Basically it looks at `this.props.experiment.error` and `this.state.enabled` to decide whether to show that element.